tools/c7n_azure/c7n_azure/resources/cosmos_db.py

```diff
@@ -53,6 +75,214 @@ class resource_type(ArmResourceManager.resource_type):
         resource_type = 'Microsoft.DocumentDB/databaseAccounts'
 
 
+class CosmosDBChildResource(ChildResourceManager):
+
+    class resource_type(ChildTypeInfo):
+        parent_spec = ('cosmosdb', True)
+        parent_manager_name = 'cosmosdb'
+        raise_on_exception = False
+        annotate_parent = True
+
+    @staticmethod
+    @lru_cache()
+    def get_cosmos_key(resource_group, resource_name, client):
+        key_result = client.database_accounts.get_read_only_keys(
+            resource_group,
+            resource_name)
+        return key_result.primary_readonly_master_key
+
+    def get_data_client(self, parent_resource):
+        key = CosmosDBChildResource.get_cosmos_key(
+            parent_resource['resourceGroup'],
+            parent_resource.get('name'),
+            self.get_parent_manager().get_client())
+        data_client = CosmosClient(
+            url_connection=parent_resource.get('properties').get('documentEndpoint'),
+            auth={'masterKey': key})
+        return data_client
+
+
+@resources.register('cosmosdb-database')
+class CosmosDBDatabase(CosmosDBChildResource):
+    """CosmosDB Database Resource
+
+    :example:
+    This policy will enumerate all cosmos databases
+
+    .. code-block:: yaml
+
+        policies:
+          - name: cosmosdb-database
+            resource: azure.cosmosdb-database
+
+    """
+
+    def enumerate_resources(self, parent_resource, type_info, **params):
+        data_client = self.get_data_client(parent_resource)
+
+        try:
+            databases = list(data_client.ReadDatabases())
+        except HTTPFailure as e:
+            if e.status_code == 403:
+                log.error("403 Forbidden. Ensure identity has `Cosmos DB Account Reader` or"
+                          "`DocumentDB Accounts Contributor` and that firewall is not "
+                          "blocking access.")
+            else:
+                log.error(e)
+            return None
```

Comment:
I'd prefer to raise an exception instead of return None (you already have try\except code to handle it..).
If you raise, no need in `log.error(e)` because it will be logged: https://github.com/cloud-custodian/cloud-custodian/pull/4305/files#diff-e447a10d22ff9e696b89d64fbdbc7ff1R106